src/rsa.py

```diff
@@ -80,45 +80,57 @@ def sign(body, user):
         except Exception as ex:
             return {"error": f"Failed to base64-decode payload: {ex}"}, 400
 
-        tmpinput = tempfile.mkstemp(prefix="rsa-")[1]
-        with open(tmpinput, 'wb') as ifile:
-            ifile.write(payload)
-            tmpfile = tempfile.mkstemp(prefix="rsa-")[1]
-            cmd = [ "openssl", "dgst", "-sign",
-                   os.path.join(RSA_DIR, key_id, "private.pem"),
-                   "-keyform", "PEM",
-                   "-sha256", "-out", tmpfile, tmpinput ]
-            cmd_result = subprocess.run(cmd, check=False)
-            if cmd_result.returncode != 0:
-                raise RuntimeError("Failed to sign payload")
-            tmpout = tempfile.mkstemp(prefix="rsa-")[1]
-            with open(tmpout, 'a', encoding='utf-8') as output:
-                output.write(sha256sum(tmpinput))
-                output.write("\n")
-                output.write("ts: " +
+        with tempfile.NamedTemporaryFile(prefix="rsa-", delete=False,
+                                         mode='wb') as tmpinput:
+            tmpinput.write(payload)
+            tmpinput_path = tmpinput.name
+
+        try:
+            with tempfile.NamedTemporaryFile(prefix="rsa-", delete=False,
+                                             mode='wb') as tmpfile:
+                tmpfile_path = tmpfile.name
+
+                cmd = [ "openssl", "dgst", "-sign",
+                       os.path.join(RSA_DIR, key_id, "private.pem"),
+                       "-keyform", "PEM",
+                       "-sha256", "-out", tmpfile_path, tmpinput_path ]
+                subprocess.run(cmd, check=True)
```

Comment:
Here `tmpfile` is open for writing from 2 places at the same time for no reason - the `with` holds the `wb` descriptor until the end of the block and the `openssl` subprocess will also write into it.

`cmd = ` and `subprocess.run` can both be one tab back.